src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -434,23 +434,10 @@ private bool BindToMethodInfo(object? target, IRuntimeMethodInfo method, Runtime
 
         private static MulticastDelegate InternalAlloc(RuntimeType type)
         {
-            MulticastDelegate? d = null;
-            InternalAlloc(new QCallTypeHandle(ref type), ObjectHandleOnStack.Create(ref d));
-            return d!;
+            Debug.Assert(type.IsAssignableTo(typeof(MulticastDelegate)));
+            return (MulticastDelegate)RuntimeTypeHandle.InternalAlloc(type);
```

Comment:
```suggestion
            return Unsafe.As<MulticastDelegate>(RuntimeTypeHandle.InternalAlloc(type));
```

Reply:
Is this avoiding the cast caching layer and forcing through the type? Basically, avoiding all the type cast validation?

Reply:
Right.